pkg/controllers/vrpq/query_reconciler.go

```diff
@@ -71,66 +79,108 @@ func (q *QueryReconciler) Reconcile(ctx context.Context, req *ctrl.Request) (ctr
 		return res, err
 	}
 
-	return ctrl.Result{}, q.runListRestorePoints(ctx, req)
+	finder := iter.MakeSubclusterFinder(q.VRec.Client, q.Vdb)
+	pods, err := finder.FindPods(ctx, iter.FindExisting)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// find a pod to execute the vclusterops API
+	var podIP string
+	for i := range pods.Items {
+		pod := &pods.Items[i]
+		if pod.Status.Phase == podRunning {
+			for j := range pod.Status.ContainerStatuses {
+				// check NMA container ready
+				if pod.Status.ContainerStatuses[j].Ready {
+					podIP = pod.Status.PodIP
+					break
+				}
+			}
+		}
+		if podIP != "" {
```

Comment:
I don't think this is needed, There is already a `break` at line 96. You probably want to check at the end of the loop if a pod satisfying the above conditions was found.

Reply:
A break at line 96 exits just the first loop. A break here exits the second loop. I mean, if we find a pod satisfying the above conditions, we don't need to check all pods in the second loop.

Reply:
Just think, if I make it into a function, it would be better. We can return the value and avoid using break statements.